lib/src/meta.dart

```diff
@@ -99,3 +99,66 @@ void _forEachGeomInGeometryObject(
     throw Exception('Unknown Geometry Type');
   }
 }
+
+/// Callback for propEach
+typedef PropEachCallback = dynamic Function(
+    Map<String, dynamic> currentProperties, num featureIndex);
+
+/// Iterate over properties in any [geoJSON] object, calling [callback] on each
+/// iteration. Similar to Array.forEach()
+///
+/// For example:
+///
+/// ```dart
+/// FeatureCollection featureCollection = FeatureCollection(
+///   features: [
+///     point1,
+///     point2,
+///     point3,
+///   ],
+/// );
+/// propEach(featureCollection, (currentProperties, featureIndex) {
+///   someOperationOnEachProperty(currentProperties);
+/// });
+/// ```
+void propEach(GeoJSONObject geoJSON, PropEachCallback callback) {
+  var i;
+  if (geoJSON is FeatureCollection) {
+    for (var i = 0; i < geoJSON.features.length; i++) {
+      if (callback(geoJSON.features[i].properties!, i) == false) break;
+    }
+  } else if (geoJSON is Feature) {
+    callback(geoJSON.properties!, 0);
+  }
```

Comment:
The already implemented function `geomEach` handles unwanted types with an else block and throws an error, like this:

https://github.com/dartclub/turf_dart/blob/6494415188765c42f434dd01b7fb88c28222cdd0/lib/src/meta.dart#L42

Do we want to have the same behaviour for all meta functions or is there a reason why this is not throwing an error?

CC @baparham

Reply:
I saw this and thought the same thing, but then checked out turf.js's implementation and they just _do nothing_ so ... ?

I think we can do what we want, and I think it makes sense to throw an error if the geoJSON is not a `Feature` or `FeatureCollection`

I would hope that the `.js` devs coming over to dart will start to think more about silent undefined behaviors and lean more towards explicit error handling, so I suspect that this would line up more with that mentality.

Reply:
I would implement it with the else-clause with `throw Exception(...)`,
because if you are using a function called `propEach` or `featureEach` it should be clear that you are either dealing with `Feature`s or a `FeatureCollection` and that the other GeoJSON types don't have properties or features in them.